### ISSUE TITLE
Ag grid Accessibility issue

### ISSUE THREAD

**Issue by Shobhit-Training** [ag-grid/ag-grid#8034]
**I'm submitting a ...**  (check one with "x")
```
[] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Customers of AG Grid**

If you are a customer you are entitled to use the AG Grid's customer support system (powered by Zendesk). Please use that channel for guaranteed response from the AG Grid team with regards bugs, feature requests and support.

**Requesting Community Support**

If you are not a customer of AG Grid, ag-grid staff will label your issue as managed-by-the-community. This means that AG Grid staff is not going to be actively looking into it and it will get closed if inactive for more than one month. The community is welcome to help with this question/support issue.

**Providing a Reproducible Scenario**
Accepted reproducible scenarios are
- A description of the detailed steps to reproduce your behaviour in one of our examples in the docs.
- A plunker

If you decide to send us a plunker, from any example in our website use the plunker button in there to fork your own code by following the steps below:

- Select the framework that is appropriate to you from the drop-down
- Open it in plunker. (Use the button plunker in our example)
- Add your changes so that the behaviour is reproduced
- Save and Freeze the plunker(On the top left corner)
- Send us the link to the plunker(You can copy the URL from the browser)

If reporting a bug make sure to state.

Current behaviour.
Expected behaviour. If possible back this up with our docs/examples if possible

**Current behavior**
 I am using Aggrid where i have vertical and horizontal scrolls , i am getting fastpass issue when i am using Accessibility insights plugin, it is giving error 

Title: WCAG 2.1.1: Ensure elements that have scrollable content are accessible by keyboard (.ag-center-cols-viewport[role="presentation"])
Tags: Accessibility, WCAG 2.1.1, scrollable-region-focusable

Issue: Ensure elements that have scrollable content are accessible by keyboard (scrollable-region-focusable - https://accessibilityinsights.io/info-examples/web/scrollable-region-focusable)

Target application: MSX To Do - https://localhost:3001/consumptionDetails?EstimateId=B5F6C6BD-0C75-4F4D-BF49-B60C43D3B68D

Element path: div[aria-colcount="11"] > .ag-body.ag-layout-normal[role="presentation"] > .ag-body-clipper.ag-layout-normal[role="presentation"] > .ag-body-viewport.ag-row-animation.ag-layout-normal > .ag-center-cols-clipper[role="presentation"] > .ag-center-cols-viewport[role="presentation"]

Snippet: <div class="ag-center-cols-viewport" role="presentation" style="height: calc(100% + 17px);">

How to fix: 
Fix any of the following:
  Element should have focusable content
  Element should be focusable

Environment: Chrome version 125.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.42.0 (axe-core 4.8.4), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

**Expected behavior**
Expected behavior is no fastpass issue should occur 

**Please tell us about your environment:**

* **AG Grid version:** X.X.X
I am using Aggrid version 29 , i checked in your provided plugin and plunker same error is ther

* **Browser:**
 Chrome , edge
<img width="943" alt="issue aggrid" src="https://github.com/ag-grid/ag-grid/assets/73124414/812e4e65-00b2-4ed5-b574-2d5bb7674c10">
<img width="943" alt="issue aggrid" src="https://github.com/ag-grid/ag-grid/assets/73124414/5ed5be46-4a0b-44b8-b031-b17bcb0fbf2f">

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by AG-Zoheil** [comment]
Hi,
 
Thank you for submitting this.
 
The accessibility audit tools base their results on static analysis of the DOM generated by AG Grid. This dictates that an element with scrollable content are accessible by keyboard.
 
AG Grid uses its own custom container to provide row and column virtualization to allow showing an unlimited number of rows in the grid while recycling a limited number of actual DOM elements to show the visible rows. This custom container doesn't use the default focus and scrolling logic that non-virtualised elements use and that accessibility audit tools expect. In order to support virtualised lists, AG Grid manages focus and keyboard navigation using its own custom focus, scrolling and keyboard navigation code.
 
Accessibility audit tools are not aware of the custom code to handle focus, scrolling and keyboard navigation across AG Grid and flag this as an issue. However, in fact there are no issues in scrolling/focus/keyboard navigation in AG Grid.

I hope this clarifies that there is no issue here.
 
Kind regards,
Zoheil‍
